extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/DoublesSketchComplexMetricSerde.java

```
@@ -0,0 +1,98 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.quantiles;
+
+import com.yahoo.memory.Memory;
+import com.yahoo.sketches.quantiles.DoublesSketch;
+import com.yahoo.sketches.quantiles.UpdateDoublesSketch;
+import io.druid.data.input.InputRow;
+import io.druid.segment.column.ColumnBuilder;
+import io.druid.segment.data.GenericIndexed;
+import io.druid.segment.data.ObjectStrategy;
+import io.druid.segment.serde.ComplexColumnPartSupplier;
+import io.druid.segment.serde.ComplexMetricExtractor;
+import io.druid.segment.serde.ComplexMetricSerde;
+
+import java.nio.ByteBuffer;
+
+public class DoublesSketchComplexMetricSerde extends ComplexMetricSerde
+{
+
+  private static final DoublesSketchObjectStrategy strategy = new DoublesSketchObjectStrategy();
+
+  @Override
+  public String getTypeName()
+  {
+    return DoublesSketchModule.DOUBLES_SKETCH;
+  }
+
+  @Override
+  public ObjectStrategy<DoublesSketch> getObjectStrategy()
+  {
+    return strategy;
+  }
+
+  @Override
+  public ComplexMetricExtractor getExtractor()
+  {
+    return new ComplexMetricExtractor()
+    {
+      @Override
+      public Class<?> extractedClass()
+      {
+        return DoublesSketch.class;
+      }
+
+      @Override
+      public Object extractValue(final InputRow inputRow, final String metricName)
+      {
+        final Object object = inputRow.getRaw(metricName);
+        if (object instanceof String) {
+          String objectString = (String) object;
+          if (Character.isDigit((objectString).charAt(0))) {
+            try {
+              Double doubleValue = Double.parseDouble(objectString);
+              UpdateDoublesSketch sketch = DoublesSketch.builder().setK(2).build();
+              sketch.update(doubleValue);
+              return sketch;
+            }
+            catch (NumberFormatException e) {
+              // Log.info("Expected Double. Got string with value " +
```

Consider adding Log.debug saying expected was `Double` but found base64Encoded string.

---

why debug? should we use log.error? this should never happen, but it is not fatal either

---

after a discussion with @akashdw we believe that it is better to throw an exception here